email-templates/daily/html.ejs

```diff
@@ -0,0 +1,271 @@
+ * Suite 510, Berkeley, CA 94720-1620, (510) 643-7201, otl@berkeley.edu,
+ * http://ipira.berkeley.edu/industry-info for commercial licensing opportunities.
+ *
+ * IN NO EVENT SHALL REGENTS BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT, SPECIAL,
+ * INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS, ARISING OUT OF
+ * THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF REGENTS HAS BEEN ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
+ * SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF ANY, PROVIDED HEREUNDER IS PROVIDED
+ * "AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
+ * ENHANCEMENTS, OR MODIFICATIONS.
+ */
+%>
+
+<%- include('../_shared/header') %>
+
+<%
+var greeting = 'You have ';
+var parts = [];
+var activityTypeCounts = _.countBy(activities, 'type');
+if (activityTypeCounts.asset_comment) {
+  if (activityTypeCounts.asset_comment === 1) {
+    parts.push('new comments on 1 asset')
+  } else {
+    parts.push('new comments on ' + activityTypeCounts.asset_comment + ' assets')
+  }
+}
+if (activityTypeCounts.asset_comment_reply) {
+  if (activityTypeCounts.asset_comment_reply === 1) {
+    parts.push('1 reply on a comment')
+  } else {
+    parts.push(activityTypeCounts.asset_comment_reply + ' replies on your comments')
+  }
+}
+if (activityTypeCounts.whiteboard_chat) {
+  if (activityTypeCounts.whiteboard_chat === 1) {
+    parts.push('new chat messages on 1 whiteboard')
+  } else {
+    parts.push('new chat messages on ' + activityTypeCounts.whiteboard_chat + ' whiteboards');
+  }
+}
+
+if (parts.length == 1) {
+  greeting += parts[0]
+} else if (parts.length == 2) {
+  greeting += parts.join(' and ');
+} else {
+  greeting += parts[0] + ',' + parts[1] + ' and ' + parts[2]
+}
+greeting += '.';
+
+// Only show a greeting if there is more than 1 activity type
+if (parts.length > 1) {
+-%>
+
+<tr id="greeting">
+  <td>
+    <h1><%= greeting %></h1>
+  </td>
+</tr>
+
+<%
+}
+
+activities.forEach(function(activity) {
+%>
+<tr class="row activity-row">
+  <td>
+    <table cellpadding="0" cellspacing="0">
+      <tbody>
+        <tr>
+          <td>
+
+            <div class="activity-summary-container">
+              <div class="wrapped">
+                <%
+                  if (activity.type === 'asset_comment') {
+                    var assetLink = course.assetlibrary_url + '?col_asset=' + activity.asset.id
+                    var commenters = _.pluck(activity.asset.comments, 'user');
+                    var summaryActors = getSummaryActors(commenters, user);
+                -%>
+                    <%= summaryActors %> <a href="<%= assetLink %>">commented</a> on your asset ‘<a href="<%= assetLink %>"><%= activity.asset.title %></a>’.
+                <% } else if (activity.type === 'asset_comment_reply') {
```

Comment:
When someone replies to your comment on your own asset, which summary is going to win?